app/src/main/java/pl/droidevs/books/about/AboutActivity.java

```diff
@@ -0,0 +1,98 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setTitle(R.string.about);
+        setContentView(R.layout.activity_about);
+
+        viewModel = ViewModelProviders.of(this).get(AboutViewModel.class);
+
+        if (savedInstanceState != null) {
+            int selectedContributorIndex = savedInstanceState.getInt(EXTRAS_SELECTED_CONTRIBUTOR_INDEX, 0);
+            viewModel.selectContributor(selectedContributorIndex);
+        } else {
+            getSupportFragmentManager()
+                    .beginTransaction()
+                    .add(R.id.fl_about, AboutVPFragment.newInstance(), AboutFragment.TAG)
+                    .commit();
+        }
+    }
+
+    @Override
+    protected void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+        outState.putInt(EXTRAS_SELECTED_CONTRIBUTOR_INDEX, viewModel.getSelectedContributorIndex().getValue());
+    }
+
+    public void showContributorDetails(ImageView imageView, ConstraintLayout constraintLayout, int index) {
+
+        int FADE_DEFAULT_TIME = 250;
+        int MOVE_DEFAULT_TIME = 500;
+
+        Fragment previousFragment = getSupportFragmentManager().findFragmentById(R.id.fl_about);
+        Fragment nextFragment = ContributorFragment.newInstance();
+        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction()/*.setReorderingAllowed(true)*/;
+
+        Fade exitFade = new Fade();
+        exitFade.setDuration(FADE_DEFAULT_TIME);
+        previousFragment.setExitTransition(exitFade);
+
+        TransitionSet enterTransitionSet = new TransitionSet();
+        enterTransitionSet.addTransition(TransitionInflater.from(this).inflateTransition(android.R.transition.move));
+        enterTransitionSet.addTransition(TransitionInflater.from(this).inflateTransition(android.R.transition.explode));
+        enterTransitionSet.setDuration(MOVE_DEFAULT_TIME);
+        enterTransitionSet.setStartDelay(FADE_DEFAULT_TIME);
+        nextFragment.setSharedElementEnterTransition(enterTransitionSet);
+
+        Fade enterFade = new Fade();
+//        android.support.transition.
+//        enterFade.setStartDelay(MOVE_DEFAULT_TIME + FADE_DEFAULT_TIME);
+        enterFade.setDuration(MOVE_DEFAULT_TIME + FADE_DEFAULT_TIME);
+        nextFragment.setEnterTransition(enterFade);
+
+        transaction.addSharedElement(imageView, imageView.getTransitionName());
+        transaction.addSharedElement(constraintLayout, constraintLayout.getTransitionName());
+        transaction.replace(R.id.fl_about, nextFragment, ContributorFragment.TAG);
+        transaction.addToBackStack(null).commitAllowingStateLoss();
+    }
+
+    @Override
+    public void onBackPressed() {
+        ContributorFragment myFragment = (ContributorFragment) getSupportFragmentManager().findFragmentByTag(ContributorFragment.TAG);
+        if (myFragment != null && myFragment.isVisible()) {
```

Comment:
if for if instance of. Kod aż woła o refaktor. Trudno określić co tutaj faktycznie się dzieje.

Reply:
Fakt. Poprawiam to właśnie.